### ISSUE TITLE
NgbModal and Upgraded Component does not work

### ISSUE THREAD

**Issue by schmitch** [ng-bootstrap/ng-bootstrap#1758]
Please remember, the issues forum is __NOT__ for support requests. It is for bugs and feature requests only.
Please read https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md and search
existing issues (both open and closed) prior to opening any new issue and ensure you follow the instructions therein.

### Bug description:
Currently when opening a Modal with an Upgraded AngularJS 1.x component one get the following error:
```
Modal Error: No provider for $scope!
    at injectionError (core.es5.js:1169)
    at noProviderError (core.es5.js:1207)
    at ReflectiveInjector_.webpackJsonp.../../../core/@angular/core.es5.js.ReflectiveInjector_._throwOrNull (core.es5.js:2649)
    at ReflectiveInjector_.webpackJsonp.../../../core/@angular/core.es5.js.ReflectiveInjector_._getByKeyDefault (core.es5.js:2688)
    at ReflectiveInjector_.webpackJsonp.../../../core/@angular/core.es5.js.ReflectiveInjector_._getByKey (core.es5.js:2620)
    at ReflectiveInjector_.webpackJsonp.../../../core/@angular/core.es5.js.ReflectiveInjector_.get (core.es5.js:2489)
    at resolveNgModuleDep (core.es5.js:9475)
    at NgModuleRef_.webpackJsonp.../../../core/@angular/core.es5.js.NgModuleRef_.get (core.es5.js:10557)
    at Object.resolveDep (core.es5.js:11060)
    at Injector_.webpackJsonp.../../../core/@angular/core.es5.js.Injector_.get (core.es5.js:10298)
```

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: 4.3.4

ng-bootstrap: 1.0.0-beta.1

Bootstrap: 3.3.7 (actually we made the modal to work by some additional scss, the modal correctly opens, but as soon as we drop the upgraded component in, we get the error above).

Currently the Upgraded Component is only a small component that provides the datepicker of angular-ui-bootstrap, it works in every component, expect NgbModal components.
Sadly NgbDatepicker did not work on Bootstrap 3, so we created a Component and Downgrade it, however that is not possibly.

**Comment by pkozlowski-opensource** [comment]
Hi! We need a plunker for all bug reports: https://github.com/ng-bootstrap/ng-bootstrap/blob/master/README.md#you-think-youve-found-a-bug

**Comment by schmitch** [comment]
Hm, I tried to plunkerize.
However I had problems en masse.

Then I found: https://plnkr.co/edit/qLndK04TuPXeYxC9WNFg?p=preview
Which has something like:

```
  providers: [
    {
      provide: '$scope',
      useFactory: i => i.get('$rootScope'),
      deps: ['$injector']
    }
  ],
```
After adding that to my code, somehow even the modal started to work.
I would guess that angular-cli somehow managed to make it work without Modals, but once inside I opened a modal, it couldn't assosciate it.
